### PR TITLE
Add example for distributed learning

### PR DESCRIPTION
**Pull request by andydiwenzhu**
Signed-off-by: Diwen Zhu <diwen.zdw@alibaba-inc.com>

What do these changes do?
-------------------------

This example demonstrates a distributed learning example using dask + horovod + v6d, which extends a real-world keras example with few lines of modifications.

Related issue number
--------------------

Fixes #issue number

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -36,6 +36,12 @@
    notes/dask.rst
    notes/ml.rst
 
+.. toctree::
+   :maxdepth: 6
+   :caption: Examples
+   
+   notes/distibuted_learning.rst
```
Comment (sighingnow) [inline docs/index.rst:43]: Its ok to put it under the "Thirt party integration" chapter, IMO.

Just after the `ml.rst`.

File: docs/notes/distributed_learning.rst [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+10 numerical features and 2 categorical features.
+Each instance is categorized into 1 of 7 classes.
+
+The solution contains two steps:
+
+1) preprocess the data in pandas to extract the 12 features and the label
+
+2) define and train the model in keras
+
+3) the preprocessed data is stored in files
```
Comment (sighingnow) [inline docs/notes/distributed_learning.rst:41]: Use rst's list instead.

File: docs/notes/distributed_learning.rst [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+An Example from Keras
+---------------------
+This example_ uses the Covertype dataset from the UCI Machine Learning Repository.
+The task is to predict forest cover type from cartographic variables.
+The dataset includes 506,011 instances with 12 input features:
+10 numerical features and 2 categorical features.
+Each instance is categorized into 1 of 7 classes.
+
+The solution contains two steps:
```
Comment (sighingnow) [inline docs/notes/distributed_learning.rst:35]: Should be `three`?

File: docs/notes/distributed_learning.rst [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+3) the preprocessed data is stored in files
+
+Mapping the solution to distributed learning, we have:
+
+1) preprocess the data in dask.dataframe
+
+2) train the model in horovod.keras
+
+3) share the preprocessed data using Vineyard
```
Comment (sighingnow) [inline docs/notes/distributed_learning.rst:49]: Ditto.

File: docs/notes/distributed_learning.rst [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+Then we can execute the distributed training with the command:
+
+.. code:: bash
+
+    horovodrun -np 4 -H h1:1,h2:1,h3:1,h4:1 python train.py o00d60aba46eaf536
+
+All the other parts of training procedure are the same as the single machine solution.
+
+Conclusion
+----------
```
Comment (sighingnow) [inline docs/notes/distributed_learning.rst:237]: Leave a blank line between `-------` and the content.

File: docs/index.rst (hunk 1 above)
Comment (andydiwenzhu) [reply docs/index.rst:43]: I would like to have an independent example directory where real-world examples are placed. While the 3rd party integration directory focuses on "Integration". Do you think it's OK?

**Comment by sighingnow** [comment]
In favour of #487.

**Pull request closed without merging**